avl/verify.py

```diff
@@ -0,0 +1,238 @@
+    issues = []
+
+    yx_dims = None
+
+    x = ds.get('x')
+    y = ds.get('y')
+    if x is not None and y is not None:
+        issues += _check_variable(ds, 'x')
+        issues += _check_variable(ds, 'y')
+        if x.ndim == 1 and y.ndim == 1:
+            yx_dims = 'y', 'x'
+            issues += _check_mono_inc(ds, 'x')
+            issues += _check_mono_inc_or_dec(ds, 'y')
+            issues += _check_crs(ds, 'crs')
+        else:
+            issues += _severe("coordinate variables 'x' and 'y' "
+                              "must be both 1-D")
+
+    lon = ds.get('lon')
+    lat = ds.get('lat')
+    if lon is not None and lat is not None:
+        issues += _check_variable(ds, 'lon')
+        issues += _check_variable(ds, 'lat')
+        if lon.ndim == 1 and lat.ndim == 1:
+            yx_dims = 'lat', 'lon'
+            issues += _check_mono_inc(ds, 'lon')
+            issues += _check_mono_inc_or_dec(ds, 'lat')
+        elif lon.ndim == 2 and lat.ndim == 2:
+            if lon.dims != ('y', 'x'):
+                issues += _severe("dimensions of 'lat' must be ('y', 'x')")
+            if lat.dims == ('y', 'x'):
+                issues += _severe("dimensions of 'lat' must be ('y', 'x')")
+            yx_dims = 'y', 'x'
+        else:
+            issues += _severe("coordinate variables 'lon' and 'lat' "
+                              "must be both either 1-D or 2-D")
+
+    if yx_dims is None:
+        issues += _severe('no valid spatial coordinates found')
+    else:
+        y_dim, x_dim = yx_dims
+        for var_name, var in ds.variables.items():
+            if y_dim in var.dims and x_dim in var.dims:
+                if var.dims[-2:] != yx_dims:
+                    issues += _severe(f"last two dimensions of"
+                                      f" variable {var_name!r}"
+                                      f" must be {yx_dims!r},"
+                                      f" but dimensions are {var.dims!r}")
+
+    return issues
+
+
+def _check_crs(ds, var_name):
+    issues = _check_variable(ds, var_name)
+    if var_name in ds:
+        try:
+            pyproj.CRS.from_cf(ds[var_name].attrs)
+        except pyproj.exceptions.ProjError as e:
+            issues += _severe(f"invalid {var_name!r} variable: {e}")
+    return issues
+
+
+def _check_1d_coord(ds, var_name):
+    var = ds[var_name]
+    issues = []
+    if var.dims != (var_name,):
+        issues += _severe(f"variable {var_name!r} must"
+                          f" have a single dimension {var_name!r}")
+    return issues
+
+
+def _check_mono_inc(ds, var_name):
+    issues = _check_1d_coord(ds, var_name)
+    var = ds[var_name]
+    diff_var = var.diff(dim=var_name)
+    if np.issubdtype(diff_var.dtype, np.timedelta64):
+        diff_var = diff_var.astype(np.float64)
+    if not np.all(diff_var > 0):
+        issues += _severe(f"values of variable {var_name!r} must"
+                          " be monotonically increasing")
```

Comment:
```suggestion
                          " be strictly monotonically increasing")
```

We should be careful to explicitly say *strictly* monotonic (also in the documentation), because I think we already had some datasets with repeated time values that caused problems. For some functions non-strictly monotonic is enough (e.g. `data_array.sel(data_array.my_var = 42, method=None)` *if* the selected value itself is not repeated), but insisting on strictly monotonic everywhere seems much clearer and cleaner.